File: android/src/main/java/com/tekartik/sqflite/Constant.java
package com.tekartik.sqflite;

/**
 * Constants between dart & Java world
 */

public class Constant {

    static final public String METHOD_GET_PLATFORM_VERSION = "getPlatformVersion";
    static final public String METHOD_GET_DATABASES_PATH = "getDatabasesPath";
    static final public String METHOD_DEBUG_MODE = "debugMode";
    static final public String METHOD_OPTIONS = "options";
    static final public String METHOD_OPEN_DATABASE = "openDatabase";
    static final public String METHOD_CLOSE_DATABASE = "closeDatabase";
    static final public String METHOD_INSERT = "insert";
    static final public String METHOD_EXECUTE = "execute";
    static final public String METHOD_QUERY = "query";
    static final public String METHOD_UPDATE = "update";
    static final public String METHOD_BATCH = "batch";

    static final String PARAM_ID = "id";
    static final String PARAM_PATH = "path";
    // when opening a database
    static final String PARAM_READ_ONLY = "readOnly"; // boolean

    static final String PARAM_QUERY_AS_MAP_LIST = "queryAsMapList"; // boolean

    public static final String PARAM_SQL = "sql";
    public static final String PARAM_SQL_ARGUMENTS = "arguments";
    public static final String PARAM_NO_RESULT = "noResult";

    // in batch
    static final String PARAM_OPERATIONS = "operations";
    // in each operation
    public static final String PARAM_METHOD = "method";

    static final String SQLITE_ERROR = "sqlite_error"; // code
    static final String ERROR_BAD_PARAM = "bad_param"; // internal only
    static final String ERROR_OPEN_FAILED = "open_failed"; // msg
    static final String ERROR_DATABASE_CLOSED = "database_closed"; // msg

    // memory database path
    static final String MEMORY_DATABASE_PATH = ":memory:";

    // android log tag
    static public String TAG = "Sqflite";
}

File: sqflite/lib/src/constant.dart
// Method to use
const String methodInsert = "insert";
const String methodBatch = "batch";
const String methodDebug = "debug";
const String methodOptions = "options";
const String methodCloseDatabase = "closeDatabase";
const String methodOpenDatabase = "openDatabase";
const String methodExecute = "execute";
const String methodUpdate = "update";
const String methodQuery = "query";
const String methodGetPlatformVersion = "getPlatformVersion";
const String methodGetDatabasesPath = "getDatabasesPath";
const String methodDatabaseExists = "databaseExists";
const String methodDeleteDatabase = "deleteDatabase";

// For batch
const String paramOperations = "operations";
// if true the result of each batch operation is not filled
const String paramNoResult = "noResult";
// if true all the operation in the batch are executed even if on failed
const String paramContinueOnError = "continueOnError";

// For each operation
const String paramMethod = "method";
// For each operation reponse
const String paramResult = "result";
const String paramError = "error";
const String paramErrorCode = "code";
const String paramErrorMessage = "message";
const String paramErrorData = "data";

// Result for open if a single instance was recovered from the native world
const String paramRecovered = "recovered";
// Result for open if a single instance was recovered from the native world while in a transaction
const String paramRecoveredInTransaction = "recoveredInTransaction";
// The database path (string)
const String paramPath = "path";
// The database instance name (string)
const String paramName = "name";
// The database version (int)
const String paramVersion = "version";
// The database id (int)
const String paramId = "id";
// True if the database is in a transaction
const String paramInTransaction = "inTransaction";
// When opening the database (bool)
const String paramReadOnly = "readOnly";
// When opening the database (bool)
const String paramSingleInstance = "singleInstance";

const String paramTable = "table";
const String paramValues = "values";

// for SQL query
const String paramSql = "sql";
const String paramSqlArguments = "arguments";

// Error
const String sqliteErrorCode = "sqlite_error";

const String inMemoryDatabasePath = ":memory:";

// Non final for changing it during testing
// If a database called is delayed by this duration, a print will happen
const Duration lockWarningDuration = Duration(seconds: 10);

/// No logs
final int sqfliteLogLevelNone = 0;

/// Log native sql commands
final int sqfliteLogLevelSql = 1;

/// Log native verbose
final int sqfliteLogLevelVerbose = 2;

// deprecated since 1.1.6
// @deprecated
const String methodSetDebugModeOn = "debugMode";
